eventuate-core/src/test/scala/com/rbmhtechnology/eventuate/EventsourcedActorTestKit.scala

```diff
@@ -0,0 +1,295 @@
+
+package com.rbmhtechnology.eventuate
+
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.actor.Props
+import akka.testkit.TestActorRef
+import akka.testkit.TestKit
+import akka.testkit.TestProbe
+import akka.util.Timeout
+import com.rbmhtechnology.eventuate.DurableEvent
+import com.rbmhtechnology.eventuate.EventSourcedActorTestKit.Persist
+import com.rbmhtechnology.eventuate.EventsourcedActor
+import com.rbmhtechnology.eventuate.EventsourcingProtocol.WriteSuccess
+import com.rbmhtechnology.eventuate.EventsourcingProtocol._
+import com.typesafe.config.ConfigFactory
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.FeatureSpecLike
+import org.scalatest.Matchers
+import org.scalatest.words.ShouldVerb
+
+import scala.concurrent.Await
+import scala.concurrent.Awaitable
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+object EventSourcedActorTestKit {
+
+  /** actor system all actors under test run in */
+  val actorSystem = ActorSystem("test", ConfigFactory.parseString("eventuate.log.write-timeout = 1s"))
+
+  /** Simulate persisting 'event'. Actors this is sent to must implement [[ExplicitOnEventCall]] */
+  case class Persist(event: Any)
+
+}
+
+/**
+  * Actor tests need to implement this trait as well as [[akka.testkit.TestKit]]. A default [[ActorSystem]] is given
+  * by companion object.  Example for usage of this trait: <br>
+  *
+  * {{{ class MyActorTest extends TestKit(ActorTest.actorSystem)  with ActorTest with GivenWhenThen { ... } }}}
+  *
+  * Usage: <br>
+  * - instantiate actors to test via initEventSourcedActorAsTestActorRef or initEventSourcedActor. Former returns an
+  * [[TestActorRef]] and thus allows to test _.underlyingActor. <br>
+  * - either tell returned [[ActorRef]]/[[TestActorRef]] a command or explicitly call onEvent via #callOnEvent <br>
+  * - query 'peristed' events via #expectEvent or #expectEvents <br><br>
+  *
+  * All actor's under test must have #eventLog as their eventLog. Provided event log can be viewed as a stash. Events
+  * persisted by actors using provided eventLog are stashed. By calling #expectEvent or #expectEvents one or several
+  * events are popped from stash while asserting equality to user defined output. <br>
+  * By calling #cleanEventLog the whole stash can be cleaned. <br>
+  * If eventLogs content should not be asserted, but simply queried #queryEventlog can be used. Cleans the whole stash
+  * as well.
+  *
+  * fixme: move to reasonable package.
+  */
+trait EventSourcedActorTestKit
+  extends FeatureSpecLike with BeforeAndAfterAll with BeforeAndAfterEach with ShouldVerb with Matchers {
+  this: TestKit =>
+
+  import EventSourcedActorTestKit._
+
+  /** defines duration to wait, when expecting events on event log */
+  private val expectEventTimeOut = 100 millisecond
```

Comment:
This might be problematic when running parallel to load and integration tests. Should be set to 3 seconds (after having solved the timeout issues with `queryAllEvents` and `cleanEventLog`)